nzshm_common/location/location.py

```diff
@@ -114,24 +114,23 @@ def get_locations(locations: Iterable[str], resolution: float = 0.001) -> List[C
         coded_locations: a list of coded locaitons
     """
     coded_locations: List[CodedLocation] = []
-    for location_spec in locations:
-        if '~' in location_spec:
-            lat, lon = location_spec.split('~')
+    for loc_id in locations:
+        location_id = str(loc_id)
+        if Path(location_id).exists():
+            coded_locations += _load_csv(Path(location_id), resolution)
```

Comment:
Given CSVs can have a number of use cases, it might be worth splitting this out into a separate function. (It also means we aren't checking the filesystem against every list element, whether it's appropriate or not.)

Suggest we split this out into another `get_locations_csv` function that can handle e.g. a Path, a string blob of CSV, a file handler for versatility and specific error handling.